pkg/cache/partition_info_test.go

```diff
@@ -39,6 +39,12 @@ partitions:
         - name: default
 `
 
+func createDefaultPartitionInfo(t *testing.T) *PartitionInfo {
```

Comment:
I think this way, we will lose the line of number where it failed.
There are multiple places calling `createDefaultPartitionInfo`, any of them failed, the output will only contain the line number in this function, then we are kind of lose tracking of which place causes this failure.
Can we do not pass `testing.T` in the parameter, let this function return an error and move the `assert.NilError` out of this helper function.

Reply:
Which takes us back to what we had already, just replace `if err` with the assert. There is no need to add this extra function call: it does not make it any more readable and it does not reduce the lines of code.

Reply:
make sure that when this change is reversed that the err variable is not shadowed like the change in `TestAddNewNode`